src/target/adiv5.c

```diff
@@ -94,16 +94,22 @@ typedef enum cid_class {
 #ifdef ENABLE_DEBUG
 /* The reserved ones only have an R in them, to save a bit of space. */
 static const char *const cidc_debug_strings[] = {
-	[cidc_gvc] = "Generic verification component",            /* 0x0 */
-	[cidc_romtab] = "ROM Table",                              /* 0x1 */
-	[0x2 ... 0x8] = "R",                                      /* 0x2 - 0x8 */
-	[cidc_dc] = "Debug component",                            /* 0x9 */
-	[0xa] = "R",                                              /* 0xa */
-	[cidc_ptb] = "Peripheral Test Block",                     /* 0xb */
-	[0xc] = "R",                                              /* 0xc */
-	[cidc_dess] = "OptimoDE Data Engine SubSystem component", /* 0xd */
-	[cidc_gipc] = "Generic IP component",                     /* 0xe */
-	[cidc_sys] = "Non STD System component",                  /* 0xf */
+	[cidc_gvc] = "Generic verification component",            /* 0x00 */
+	[cidc_romtab] = "ROM Table",                              /* 0x01 */
+	[0x2] = "R",                                              /* 0x02 */
```

Comment:
Do we know from the spec or the architecture manuals etc what these "R" entries should be? Or even, should they be valid strings rather than NULL for now to make it more obvious when the code hits one of these unknown/reserved CID values? The comment talks about saving space but having the table store null pointers would save more and be more obvious if we see one in the wild.

Reply:
In looking into this I refactored `adiv5_ap_type`, which seemed to have a mistake, the function could look up from indexes 0 to 7, but only 6 entries existed

Reply:
The structure of adiv5_ap_type() was built around this table from the ADIv5 spec:
![image](https://github.com/blackmagic-debug/blackmagic/assets/691140/0326e870-cb2b-433c-8582-b98646359686)

This is why the table and function was laid out how it was.

Reply:
Ah! I looked for that table but it's not in the ADIv5.2 spec I was looking at, or at least not under that name (or maybe I just missed it)

But there was in fact a problem (not the missing `> 0` check), the table was doing a simple lookup for `type -1` but the type `0x3` does not exist so the table was missing one empty entry, every type after `0x3` would have been showing the incorrect string, and type `8` would be out of bounds

I corrected the values in my commit

Reply:
Oh, great catch! that apparently completely blind-sided us. We'll re-review with that in mind